### PR TITLE
Add support for repository actions access level / permission

### LINKED ISSUE #2575: Support for repository actions permissions access
Looks like this was added to github.com around [March 2022][1] and Enterprise Server in [May 2022][2].  
It allows other repositories in the same organization or enterprise to access actions in said repository

Endpoints:  
[`GET` `/repos/{owner}/{repo}/actions/permissions/access`](https://docs.github.com/en/enterprise-cloud@latest/rest/actions/permissions#get-the-level-of-access-for-workflows-outside-of-the-repository)  
[`PUT` `/repos/{owner}/{repo}/actions/permissions/access`](https://docs.github.com/en/enterprise-cloud@latest/rest/actions/permissions#set-the-level-of-access-for-workflows-outside-of-the-repository)

I imagine this should be added as `repos_actions_permissions_access.go` 🤔 

#### Additional Links & References
https://github.com/github/roadmap/issues/74  
https://docs.github.com/en/enterprise-cloud@latest/actions/creating-actions/sharing-actions-and-workflows-with-your-enterprise  

[1]: https://github.blog/changelog/2022-03-04-sharing-github-actions-within-your-enterprise-is-now-ga/
[2]: https://docs.github.com/enterprise-server@3.5/admin/release-notes#sharing-github-actions-within-your-enterprise-is-generally-available

### PR DESCRIPTION
**Pull request by s4nji**
This PR adds support for the following endpoints and closes #2575:  
[`GET` `/repos/{owner}/{repo}/actions/permissions/access`](https://docs.github.com/en/enterprise-cloud@latest/rest/actions/permissions#get-the-level-of-access-for-workflows-outside-of-the-repository)  
[`PUT` `/repos/{owner}/{repo}/actions/permissions/access`](https://docs.github.com/en/enterprise-cloud@latest/rest/actions/permissions#set-the-level-of-access-for-workflows-outside-of-the-repository)

### CODE REVIEW

File: github/repos_actions_access.go
Comment (gmlewis) [inline github/repos_actions_access.go:55]:
```suggestion
	return s.client.Do(ctx, req, nil)
```

File: github/repos_actions_access.go
Comment (gmlewis) [inline github/repos_actions_access.go:19]:
```suggestion
	// Possible values are: "none", "organization" "enterprise".
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @s4nji !
Just a couple minor tweaks, please, then we will be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Review by gmlewis — APPROVED** [review]
Thank you, @s4nji !
LGTM.
Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
